### DISCUSSION TITLE
Mutation returning 403 Forbidden not being caught in .catch()

### DISCUSSION THREAD

**Discussion by bggolding** [reduxjs/redux-toolkit#1563] [General]
In my application for a project that runs chess tournaments, I have a little snippet that looks like:

```
         updateTournamentMatch({
                tournamentId: tournament.id,
                matches: changedMatches,
                matchPayloadFilter: api.updateMatchRePairFilter
         })
                .unwrap()
                .then(() => onClose())
                .catch((error) => setErrorMessage(error.data.message));
```
In my case the server responds with a 403 (Forbidden) error because the user doesn’t have permission to alter the tournament, the .catch() case isn’t executed and the error goes missing. I can see in the debug console that there’s a reject action dispatched. When I change the server to return 400 instead, the code works perfectly. I’m confused.

**Comment by phryneas** [comment]
That should not be the case. Do you have a custom `baseQuery` that maybe handles a 403 as a non-error case? Can you create a reproduction?

**Comment by bggolding** [comment]
My baseQuery implementation is very straightforward and largely copied from the docs:
```
    baseQuery: fetchBaseQuery({
        baseUrl: '/api',
        prepareHeaders: (headers, {getState}) => {
            // By default, if we have a token in the store, let's use that for authenticated requests
            const jwtToken = (getState() as RootState).login.jwtToken;
            if (jwtToken) {
                headers.set('Authorization', `Bearer ${jwtToken}`);
            }
            return headers;
        }
    }),
```

**Comment by bggolding** [comment]
I tried reworking the backend to change the way that it raises exceptions and with a few tweaks it has started working.

I note that the difference in the response from the server where it doesn’t work is:
```
{
  "code": 403, 
  "message": "", 
  "status": "Forbidden"
}
```
and where it does work:
```
{
  "code": 403, 
  "message": "Only staff may update tournament matches", 
  "status": "Forbidden"
}
```
I went back and changed my working error response to return an empty string and it stopped working. Very strange.
